modules/programs/yt-dlp.nix

```
@@ -6,6 +6,13 @@ let
 
   cfg = config.programs.yt-dlp;
 
+  renderedSettings = settings:
+    concatStringsSep "\n" (mapAttrsToList (name: value:
+      if (isBool value) then
+        if value then "--${name}" else ""
```

Looking closer at the yt-dlp man page, there are some Boolean options that have a `--no-` prefix. For example, `--embed-subs` and `--no-embed-subs`. So perhaps this should be

```suggestion
        if value then "--${name}" else "--no-${name}"
```

Some Boolean options don't seem to have a `--no`-variant, however, such as `--extract-audio`. But I guess for now we can just assume that people shouldn't set those to `false` 🙂 The alternative is to allow `null` and make `null` valued attributes result in nothing in the generated configuration.

---

I implemented your suggestion but did not understand the point of allowing `null` if it is for making `null` valued attributes result in nothing in the configuration.